packages/medusa/src/api-v2/admin/inventory-items/middlewares.ts

```diff
@@ -55,6 +51,21 @@ export const adminInventoryRoutesMiddlewares: MiddlewareRoute[] = [
       ),
     ],
   },
+  {
+    method: ["GET"],
+    matcher: "/admin/inventory-items/:id/location-levels",
+    middlewares: [
+      transformQuery(
+        AdminGetInventoryItemsItemLocationLevelsParams,
+        QueryConfig.listLocationLevelsTransformQueryConfig
+      ),
+    ],
+  },
+  {
+    method: ["POST"],
+    matcher: "/admin/inventory-items/:id/location-levels",
+    middlewares: [transformBody(AdminPostInventoryItemsItemLocationLevelsReq)],
```

Comment:
As we've discussed in eng-core, We want to return the entire parent object, so we need to add the `transformQuery` middleware here as well (and then in the handler refetch the item and return it)

Reply:
I've not missed the discussion in eng-core. I think returning the inventory-item invalidates the point of this endpoint (listing a sub-entity directly) as a retrieve for the inventory-item id with expand parameters would accomplish the same thing. 

I'm fine with closing this pr instead and just relying on that endpoint instead

Reply:
We only return the parent object on POST requests, so this should only be updated here.

The current state of the PR would also introduce a breaking change, as we are returning the parent entity in this endpoint in V1, and we might as well avoid that.